### ISSUE TITLE
Input issue of the trained model

### ISSUE THREAD

**Issue by Gang-Cheng-EE** [TU-Delft-AI-Energy-Lab/Deep-Statistical-Solver-for-Distribution-System-State-Estimation#5]
Dear TU Delft AI Energy Lab researchers,

Thank you for your amazing paper and helpful open-source codes. That is so useful in reproducing the project. I encountered one issue on the input variables of the trained model.

For the trained model (such as the trained mode in the folder saved_model>> cigre_meas1_dss), the **input variables are (a_flat,b_flat,U_flat,A0, training=False)** (see line 1086 in the case_study.py), where the labeled variable, i.e., U_flat, is included as input when I use the trained model to perform DSSE. However, when I use the procedure to **train a new model**, the **input variables of the trained model are (a_flat, b_flat, A0, training=False)** (see line 901 in the fun_dss.py), where the labeled variable, i.e., U_flat, is NOT included as input when I test the trained model.

It is well known that when using a trained model for inference (making predictions on new data), we only need to provide the input features that the model was trained on. Including the labeled variables as inputs to the trained model would be unnecessary and redundant for making predictions. Hence, I am confused with the input of the trained model (provided by the authors). Why the labeled variables (i.e., U_flat) are included when using it for predicting.

Thank you for your response in advance.

Sincerely,
Gang

**Comment by benhbb** [comment]
Dear Gang,

Thank you for your comment. U_flat was used as a reference value to initiate the model output. Indeed, it was not needed. The whole script has been now updated to use PyTorch with known GNN models from the libraries, still training with the WLS loss function in weak supervision. U_flat is not an issue anymore and I hope you will enjoy the new code.
